### DISCUSSION TITLE
Flutter MacOS Publishing Fails

### DISCUSSION THREAD

**Discussion by gvsakhil** [codemagic-ci-cd/codemagic-docs#2414] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1. Connect a github app developed using flutter
2. I have selected Automatic Code Signing Method
3. Added the correct bundle identifier

### Expected results

App should be published to app store successfully

### Actual results

```
Publish "/Users/builder/clone/macos/Pods/ObjectBox/Mac/OBXCodeGen.framework/Versions/A/Resources/Sourcery.pkg" to App Store Connect
Bundle identifier: Pixle.Sourcery
Install size: 20032KB
Version code: 1
Version: 1.0.0

Upload "/Users/builder/clone/macos/Pods/ObjectBox/Mac/OBXCodeGen.framework/Versions/A/Resources/Sourcery.pkg" to App Store Connect
{
    "tool-version": "7.006.15006",
    "tool-path": "/Applications/Xcode-15.2.app/Contents/SharedFrameworks/ContentDeliveryServices.framework/Versions/A/Frameworks/AppStoreService.framework",
    "os-version": "14.2.1",
    "product-errors": [
        {
            "code": -19000,
            "message": "No suitable application records were found. Verify your bundle identifier \u201cPixle.Sourcery\u201d is correct and that you are signed in with an Apple ID that has access to the app in App Store Connect.",
            "code-string": "ITunesSoftwareServiceApplicationMustEndWithProperExtension",
            "userInfo": {
                "NSLocalizedFailureReason": "App Store operation failed.",
                "NSLocalizedRecoverySuggestion": "No suitable application records were found. Verify your bundle identifier \u201cPixle.Sourcery\u201d is correct and that you are signed in with an Apple ID that has access to the app in App Store Connect.",
                "NSLocalizedDescription": "No suitable application records were found. Verify your bundle identifier \u201cPixle.Sourcery\u201d is correct and that you are signed in with an Apple ID that has access to the app in App Store Connect."
            }
        }
    ]
}
```

### Mac Build in CodeMagic -
```
Bundle identifier: Pixle.Sourcery
Install size: 20032KB
Version code: 1
Version: 1.0.0
```

Surprisingly in IOS everything is working fine and I am seeing following in IOS
### IOS Build in CodeMagic -
```
App name: Kuber POS
Bundle identifier: com.kuber.pos
Certificate expires: 2025-03-12T14:06:03.000+0000
Distribution type: App Store
Min os version: 13.0
Provisioned devices: N/A
Provisions all devices: No
Supported platforms: iPhoneOS
Version code: 97
Version: 1.0.4
```

### Build id (optional)

65f30c3d1719674680f5a0c4

**Comment by dtrdic** [comment]
conversation continued in Intercom chat
